### ISSUE TITLE
User unable to set notification preference

### ISSUE THREAD

**Issue by mattdricker** [hypothesis/h#7724]
Per user report https://app.hubspot.com/contacts/6291320/ticket/426220295/

Hypothesis user (see above ticket for username) notes that on the [Notifications setting page](https://hypothes.is/account/settings/notifications), their option to receive email notifications is unchecked. They can check the box, but when they navigate away from the setting page and then return, the checkbox has reverted to unchecked.

User has confirmed the following:
- Able to change other account settings (name, description, etc.) fine, and these changes stick.
- Issue occurs using multiple browsers
- Issue occurs using multiple devices/platforms
- Viewing Browser Dev Tools, Network requests when toggling the checkbox all return 200 status
- Console shows a few errors and warnings

![network requests when checking and unchecking](https://user-images.githubusercontent.com/71603629/119855083-9d3df580-bedf-11eb-9ed9-86d4455b8752.png)

![Firefox enhanced tracking protection OFF](https://user-images.githubusercontent.com/71603629/119855113-a5963080-bedf-11eb-920c-009421819ebd.png)

User tests on systems:
- Linux KDE Neon (Ubuntu 20.04)
- Windows
- Linux Mint
- Firefox 88
- Vivaldi 3.8.2259
- Chromium (latest)

We are unable to replicate behavior with any other user.

## Tasks

 * https://github.com/hypothesis/h/pull/7727
 * https://github.com/hypothesis/h/pull/7742
 * https://github.com/hypothesis/h/pull/7741

**Comment by jon-betts** [comment]
I've added an example POC PR which I think fixes this issue:

 * https://github.com/hypothesis/h/pull/7727
 
I think this proves the point and isn't a massive amount of work.

**Comment by jon-betts** [comment]
I've deployed a fix for this. This is hard to test in live as it requires a user with a missing subscription, but we've tested it extensively in dev and I've also tested that existing expected behavior is preserved in prod.

The new behavior should be that users with missing rows will get a row created upon trying to access it. This will happen the first time they:

 * Look at their notification preferences
 * The system checks their notification preferences when they get a reply

The default is **on**, to match the default you get when you create an account. So users might start getting replies right away without them having to enable it. They are of course free to disable it if they don't like it.
